### ISSUE TITLE
Document ideal build environment

### ISSUE THREAD

**Issue by dcoshea** [roelandjansen/pcmos386v501#8]
Moving my off-topic comment from issue #5 into a new issue since it warrants further discussion, and it would be nice to document this in `README.md`:

> I haven't created a virtual env to build it again myself, nor thought of a way to pull or push changes from there.

[...]

DOSBox would be very convenient since it requires little setup and it will mount a local directory as a DOS drive, so you can just set up some symlinks to your Git repository, but there's issue #6.

[Bochs](http://bochs.sourceforge.net/) will let you present a local directory to a virtual machine as a virtual VFAT filesystem ("vvfat"), as will [QEMU](https://www.qemu.org/), and they'll let you install whatever version of DOS you want, unlike DOSBox (which kind of does, but then loses the ability to mount a directory).  I tried with Bochs:
- It should hypothetically let you add an MBR image file and a bootsector image file into your directory so you can boot a DOS installation from a directory, but it didn't work for me.  I imagine that's because it's not presenting the `IO.SYS` and `MSDOS.SYS` files in my local directory as the first two files in the virtual VFAT filesystem.  I gave up on this.
- Another other option is to use a disk image for the first hard disk and vvfat for the second hard disk.  Two more options present themselves here:
  - Use these C: and D: drives normally, with the source code on D:, in which case `MAKEMOS.BAT` needs to be modified to use D: instead of C:.  There may be other files hard-coded to use C: too, I wouldn't know since I haven't tried this.
  - Copy the DOS files from C: to D:, then use `assign c=d` to redirect all access to C: to D:.  This worked for me (with MS-DOS 5).  In this case, C: merely needs to be bootable and have `ASSIGN.COM`, so it could be a tiny disk image.
- Alternatively, you could have a vvfat C: and just boot from a virtual floppy drive containing an `AUTOEXEC.BAT` that sets the PATH to include `C:\DOS5.0`, i.e. you only use the floppy because it's bootable, but don't use it after booting.  This might not be so convenient if you want to create PC-MOS bootable floppies using the VM; I don't know if that is something that one might want to do because I don't have any idea how it is done :)

Sadly the Bochs vvfat support seems to be buggy (I'm using the latest version 2.6.9) - it doesn't always write files back to your original directory, and sometimes filenames get combined for some reason, e.g. I ended up with a `SOURCES/src/mos5src/19term.sysUNTERM.PUB` file.

Perhaps a safer option is to run a network redirector and a file server, which would at least let you check file status from Git from outside the VM at the same time as the VM is running, since the above options don't allow (in the case of Bochs) or discourage (in the case of DOSBox) access to the files from the host while the guest is running. [Edit: but a network redirector alone isn't generally going to make your mount available as C: unless you boot from a floppy and have no hard disk; would a network redirector plus `assign c=d` work?]

**Comment by HStewartGit** [comment]
It has been many years ( actually 2 1/2  decades ) since I work for Software Link,  but when developing PC-MOS we actually used PC-MOS to developed the code.

I just spend hour and half going though memories of old code,  but a while back - I remember putting PC-MOS into VM on VM ware.   I not sure about build system any more -  I not sure there is any changes in this code since I left in 1992.   Probably my most memorable thing Is finding bug in IBM 486SLC and making Undocumented DOS functions work.

**Comment by roelandjansen** [comment]
hi Stewart (H),

nice you're here too ;-)

regarding the build stuff and vmware -- got quite some sizing issues whre msys failed formwat failed or hdsetup failed.

You're welcome for sure.

Roeland

ps I am waiting fpor the cdrom driver sources -- got that too to GPL.

**Comment by HStewartGit** [comment]
Roland,

I probably forgot how I did this - what I would do is boot to dos VM - make sure memory and storage - back in those day a Gig drive and 16Meg of memory was huge.  Format the drive with dos  and then but with installation floppies and installed on VM disk.

If you have too much memory or too much storage - there maybe issues with code on formatting display and stuff.  I remember fixing issue with memory in $386.sys and in directory command.

I hope this helps

Stewart

If you need more storage - serious don't believe more memory can be used - you can play with different installations and track down max you can have.

**Comment by the-grue** [comment]
I have started playing with this in parallel on both DOSBox and VirtualBox.  I have a paired Linux VirtualBox environment that has the git repo on it and it can mount the drives natively to swap files in/out.

Has anyone identified all of the tools currently required to build the repo?

I have seen Borland C++, MASM, possibly Microsoft C so far.

**Comment by stsp** [comment]
Please see #22.
Fully functional automated build.

**Comment by src153** [comment]
Recent work supersedes, closing.
